File: src/models/hotelRoom.js
'use strict';
const {
  Model
} = require('sequelize');
module.exports = (sequelize, DataTypes) => {
  class HotelRoom extends Model {
    /**
     * Helper method for defining associations.
     * This method is not a part of Sequelize lifecycle.
     * The `models/index` file will call this method automatically.
     */
    static associate(models) {
      HotelRoom.belongsTo(models.Hotel,{
        foreignKey: "hotel_id",
        targetKey: "id",
        as: "hotelromData",
      })
      HotelRoom.belongsTo(models.TypeRoom,{
        foreignKey: "type_id",
        targetKey: "id",
        as: "typeroomData",
      })
      HotelRoom.hasMany(models.Booking ,{
        foreignKey: "room_id",
        as: "hotelrombookingData",
      })
    }
  }
  HotelRoom.init({
    hotel_id : DataTypes.INTEGER, 
    type_id  : DataTypes.INTEGER,
    name : DataTypes.STRING,
    bed_quantity : DataTypes.INTEGER,
    price : DataTypes.INTEGER,
    description : DataTypes.TEXT,
    image : DataTypes.STRING,
  }, {
    sequelize,
    modelName: 'HotelRoom',
  });
  return HotelRoom;
};

File: Databases/Entities/HotelRoom.cs




using System.ComponentModel.DataAnnotations;
using HotelApp.API.Database.Entities;

namespace HotelApp.API.Databases.Entities
{
    public class HotelRoom
    {
        [Key]
        public int Id { get; set; }

        [StringLength(100)]
        public string Name { get; set; } = null!;

        public int Bed_quantity { get; set; }

        public int Price { get; set; }

        public string Description { get; set; } = null!;

        public string Image { get; set; } = null!;

        public int TypeRoomId { get; set; }
        public int HotelId { get; set; }
        public virtual TypeRoom TypeRoom { get; set; } = null!;
        public virtual Hotel Hotel { get; set; } = null!;

        public virtual ICollection<Booking> Bookings { get; set; } = new HashSet<Booking>();
    }
}